src/components/ContentLoader.tsx

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useState } from 'react'
+import { ENV } from '../lib/env'
+
+interface ContentResponse {
+  id: string
+  content: Array<{variant: string, html: string}>
+}
+
+enum FetchStatus {
+  Unfetched,
+  FetchSuccess,
+  FetchFailure
+}
+
+export const ContentLoader = ({ contentId, contentLoadedCallback }: { contentId: string, contentLoadedCallback: () => void }): JSX.Element => {
+  const [innerHTMLContent, setInnerHTMLContent] = useState<string>('')
+  const [fetchStatus, setFetchStatus] = useState<Number>(FetchStatus.Unfetched)
+
+  const fetchContent = async (): Promise<void> => {
+    const request = new Request(`${ENV.OS_RAISE_CONTENT_URL_PREFIX}/${contentId}.json`)
+
+    try {
+      const response = await fetch(request)
+      const data = await response.json() as ContentResponse
+      const htmlContent = data.content[0].html
+
+      setInnerHTMLContent(htmlContent)
+      setFetchStatus(FetchStatus.FetchSuccess)
+    } catch {
+      setFetchStatus(FetchStatus.FetchFailure)
+    }
+  }
+
+  useEffect(() => {
+    fetchContent().catch(() => {})
+  }, [])
```

Comment:
Does this `UseEffect` hook enable `fetchContent()` to be called (if so by what), capture its errors, or both? I'm curious very generally about how these hooks are used

Reply:
It's the first. The usage here is probably best described by the notes [here](https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect) on conditionally firing effects. The key thing here is providing `[]` as the dependency input so we know `fetchContent` is only invoked once.